service/lib/agama/autoyast/report_patching.rb

```diff
@@ -0,0 +1,106 @@
+# find current contact information at www.suse.com.
+
+# Goal of this file is monkey patch Popup and Report functionality to not try to use UI
+# and instead adapt it for agama needs.
+# Do not directly require agama ruby files to be able to keep autoyast converter and agama
+# independent.
+# TODO: what to do if it runs without agama? Just print question to stderr?
+
+require "yaml"
+require "yast/execute"
+
+module Yast2
+  class Popup
+    class << self
+      # Keep in sync with real Yast2::Popup
+      def show(message, details: "", headline: "", timeout: 0, focus: nil, buttons: :ok, richtext: false, style: :notice)
+        # at first construct agama question to display.
+        # NOTE: timeout is not supported.
+        # FIXME: what to do with richtext?
+        text = message
+        text += "\n\n" + details unless details.empty?
+        options = generate_options(buttons)
+        question = {
+          # TODO: id for newly created question is ignored, but maybe it will be better to not have to specify it at all?
+          id:             0,
+          class:          "autoyast.popup",
+          text:           text,
+          options:        generate_options(buttons),
+          default_option: focus || options.first,
+          data:           {}
+        }
+        data = { generic: question }.to_yaml
+        answer_yaml = Yast::Execute.locally!("agama", "questions", "ask", stdin: data, stdout: :capture)
+        answer = YAML.load(answer_yaml)
+        answer["generic"]["answer"].to_sym
+      end
+
+    private
+
+      def generate_options(buttons)
+        case buttons
+        when :ok
+          [:ok]
+        when :continue_cancel
+          [:continue, :cancel]
+        when :yes_no
+          [:yes, :no]
+        when Hash
+          buttons.keys
+        else
+          raise ArgumentError, "Invalid value #{buttons.inspect} for buttons"
+        end
+      end
+    end
+  end
+end
+
+# needed to ask for GPG encrypted autoyast profiles
+# TODO: encrypt agama profile? is it needed?
+module UI
+  class PasswordDialog
+    def new(label, confirm: false)
+      @label = label
+      # NOTE: implement confirm if needed
+    end
+
+    def run
+      # at first construct agama question to display.
+      text = @label
+      question = {
+        # TODO: id for newly created question is ignored, but maybe it will be better to not have to specify it at all?
+        id:             0,
+        class:          "autoyast.password",
+        text:           text,
+        options:        ["ok", "cancel"],
+        default_option: "cancel",
+        data:           {}
+      }
+      data = { generic: question, with_password: {} }.to_yaml
+      answer_yaml = Yast::Execute.locally!("agama", "questions", "ask", stdin: data, stdout: :capture)
+      answer = YAML.load(answer_yaml)
+      result = answer["generic"]["answer"].to_sym
+      return nil if result == "cancel"
```

Comment:
```suggestion
      return nil if result == :cancel
```

Reply:
thanks, good catch. Originally there was not to_sym, but yast popup API expects it and I forgot to update it also here.